File: back-end/src/main/java/server/model/GodinaStudijaPredmet.java
package server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GodinaStudijaPredmet {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne(optional = false)
	private GodinaStudija godinaStudija; 

    @ManyToOne(optional = false)
	private Predmet predmet;   

    @Column(nullable = false)
    private Boolean vidljiv = true;
}

File: front-end/src/app/models/GodinaStudijaPredmet.ts
import { GodinaStudija } from "./GodinaStudija";
import { Predmet } from "./Predmet";

export interface GodinaStudijaPredmet {
  id: number;
  vidljiv: boolean;
  godinaStudija: GodinaStudija;
  predmet: Predmet;       
}
